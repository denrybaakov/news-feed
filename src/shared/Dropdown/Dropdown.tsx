import React, { ReactNode, useEffect, useState } from 'react';
import styles from './dropdown.css';

interface IDropdownProps {
  children?: ReactNode,
  button: ReactNode,
  isOpen: boolean,
  onOpen?: () => void,
  onClose?: () => void
}

const VOID_FUNC = () => { }

export function Dropdown({
  children,
  button,
  isOpen,
  onOpen = VOID_FUNC,
  onClose = VOID_FUNC
}: IDropdownProps) {

  const [isDropdown, setIsDropdown] = useState(false)
  useEffect(() => setIsDropdown(false), [isOpen])
  useEffect(() => isDropdown ? onOpen() : onClose(), [isDropdown])

  const handleOpen = () => isDropdown ? setIsDropdown(!isDropdown) : null

  return (
    <div className={styles.container}>
      <div onClick={handleOpen}>
        {button}
      </div>
      {isDropdown && (
        <div className={styles.listContainer}>
          <div
            className={styles.list}
            onClick={() => setIsDropdown(false)}
          >
            {children}
          </div>
        </div>
      )}
    </div>
  )
}
