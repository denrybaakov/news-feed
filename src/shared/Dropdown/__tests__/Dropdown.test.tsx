/**
 * @jest-environment jsdom
 * */

import React from "react"
import { Dropdown } from "../Dropdown"
import { shallow } from 'enzyme'

import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

Enzyme.configure({ adapter: new Adapter() })

describe('Dropdown', () => {
  test('Рендер компонента Dropdown', () => {
    const wrapper = shallow(<Dropdown children={<div />} button={<button>Test</button>} />)
    expect(wrapper).toBeDefined()
    // expect(wrapper.find('#button')).toBeDefined()
    expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy()
  })

  test('Рендер компонента Dropdown (snapshot)', () => {
    const wrapper = shallow(<Dropdown children={<div />} button={<button>Test</button>} />)
    expect(wrapper).toMatchSnapshot()
  })
})
