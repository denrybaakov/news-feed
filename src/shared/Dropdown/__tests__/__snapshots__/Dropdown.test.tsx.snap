// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Рендер компонента Dropdown (snapshot) 1`] = `
<div
  className="container"
>
  <div
    onClick={[Function]}
  >
    <button>
      Test
    </button>
  </div>
</div>
`;
